pinot-core/src/main/java/org/apache/pinot/core/query/request/context/QueryContext.java

```diff
@@ -74,6 +77,7 @@
   private final List<ExpressionContext> _groupByExpressions;
   private final FilterContext _havingFilter;
   private final List<OrderByExpressionContext> _orderByExpressions;
+  private final Function<Class<?>, Map<?, ?>> _sharedValues = MemoizedClassAssociation.of(ConcurrentHashMap::new);
```

Comment:
nit: can we put this after _brokerRequest? to me all the private final variable here are parsed from _BrokerRequest thus "immutable", it would be nice to keep them together